homeassistant/components/tado/__init__.py

```diff
@@ -314,6 +314,9 @@ def set_zone_off(self, zone_id, overlay_mode, device_type="HEATING"):
     def set_temperature_offset(self, device_id, offset):
         """Set temperature offset of device."""
         try:
+            if template.is_template_string(str(offset)):
+                offset = template.Template(str(offset), self.hass).render()
```

Comment:
This shouldn't be needed the template will be rendered before reaching the service schema.

Reply:
Ahh ok I'll revert that commit (googles 'how to revert a commit' 😄 ), apologies... still learning the process

Its not, template doesn't work.. It passes the whole template as a string.. This code I wrote works to render that as a value, no ideas why it doesn't work 🤷 I expected, like you said, HA would render it before passing to the service....

Reply:
Ok I reverted that commit, seems to have worked. Back on commit b54baf7f31 now

Reply:
How are you testing the template and service?

Reply:
I used Developer Tools to call the service with a template. When it didn't work I increased output to the logs to see what was happening.

I didn't need a template, hence not writing it in the first place, one of the guys on the feature request pointed out it didn't work so I tested it then and found that to be true, it didn't. I kinda felt like it should so looked into it.

Happy for this PR to go in without now I reverted the code. Maybe one for the future? 

Feel free to point me at any docs or example code etc if you agree we should have it..

Reply:
Hey, sorry for delay, finally got round to testing that. Yeah it works, you were right (I know you know that 😉 )

Is there anything else to do? The tests failed for some reason when I reverted the commit, looked like a timeout. Is there a way to kick them off again?

Reply:
I've restarted CI. Let's see if it passes this time.